### PR TITLE
Implement S3 transfer acceleration support.

### LINKED ISSUE #37: Support S3 transfer acceleration
S3 transfer acceleration requires this: https://docs.aws.amazon.com/AmazonS3/latest/dev/transfer-acceleration.html#transfer-acceleration-getting-started

When trying to use this:

```
time="2018-07-07T13:57:54Z" level=info msg="Restoring cache at /fusionapp/fusion/master/archive.tar"
time="2018-07-07T13:57:54Z" level=info msg="Retrieving file in fusionapp at fusion/master/archive.tar"
time="2018-07-07T13:57:57Z" level=warning msg="Cache could not be restored The specified bucket does not exist."
```

I believe this is because it expects the first path component to be the bucket name.

### PR DESCRIPTION
**Pull request by mithrandi**
Closes #37.

I'm not sure where to update the docs for this, could someone point me in the right direction?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -117,6 +117,11 @@ func main() {
 			Usage:  "s3 server",
 			EnvVar: "PLUGIN_SERVER,PLUGIN_ENDPOINT,CACHE_S3_ENDPOINT,CACHE_S3_SERVER,S3_ENDPOINT",
 		},
+		cli.StringFlag{
+			Name:   "accelerated-endpoint",
+			Usage:  "s3 accelerated endpoint",
+			EnvVar: "PLUGIN_ACCELERATED_ENDPOINT,CACHE_S3_ACCELERATED_ENDPOINT,CACHE_S3_ACCELERATED_SERVER,S3_ACCELERATED_ENDPOINT",
```
Comment (donny-dont) [inline main.go]: Whats with all the environment variables?

File: main.go [hunk 2]
```diff
@@ -293,12 +298,15 @@ func s3Storage(c *cli.Context) (storage.Storage, error) {
...
 
+	acceleratedEndpoint := c.String("accelerated-endpoint")
+
 	// Get the access credentials
 	access := c.String("access-key")
 	secret := c.String("secret-key")
 
 	return s3.New(&s3.Options{
 		Endpoint: endpoint,
+		AcceleratedEndpoint: acceleratedEndpoint,
```
Comment (donny-dont) [inline main.go]: gofmt this

File: storage/s3/s3.go [hunk 3]
```diff
@@ -15,6 +15,7 @@ import (
 // Options contains configuration for the S3 connection.
 type Options struct {
 	Endpoint   string
+	AcceleratedEndpoint string
```
Comment (donny-dont) [inline storage/s3/s3.go]: gofmt

File: main.go (hunk 1 above)
Comment (mithrandi) [reply main.go]: I basically copied them from `--server`; which one(s) should I keep?

Comment (donny-dont) [reply main.go]: If this would ever be a secret then just keep `CACHE_S3_ACCELERATED_ENDPOINT` and remove the other two.

File: main.go (hunk 2 above)
Comment (mithrandi) [reply main.go]: Oops, done.

File: main.go (hunk 1 above)
Comment (mithrandi) [reply main.go]: I'm not sure if you would ever want this to be a secret, but I guess if `--server` is a secret then this would be unset or also a secret.

Comment (mithrandi) [reply main.go]: Removed the others.

**Comment by donny-dont** [comment]
Thanks @mithrandi ! Let me know if that works for you.

**Pull request merged**